pkg/pool-stable-phantom/test/StablePhantomPool.test.ts

```diff
@@ -968,237 +971,130 @@ describe('StablePhantomPool', () => {
         await tokens.approve({ from: lp, to: pool.vault });
       });
 
-      describe('accounting', () => {
-        const amount = fp(1);
-
-        sharedBeforeEach('update cache', async () => {
-          await pool.updateProtocolSwapFeePercentageCache();
-        });
-
-        enum AmountKind {
-          WITH_FEE,
-          WITHOUT_FEE,
-        }
+      function itAccountsForProtocolFees() {
+        describe('accounting', () => {
+          const amount = fp(1);
+          let inRecoveryMode: boolean;
 
-        function getAproxDueFee(amount: BigNumber, kind: AmountKind): BigNumber {
-          // In StablePools, BPT and underlying tokens are almost equivalent. This means that the token fee amount is a
-          // good estimate of the equivalent BPT fee amount.
+          sharedBeforeEach('update cache', async () => {
+            await pool.updateProtocolSwapFeePercentageCache();
+            inRecoveryMode = await pool.inRecoveryMode();
+          });
 
-          if (kind == AmountKind.WITHOUT_FEE) {
-            amount = amount.mul(fp(1)).div(fp(1).sub(swapFeePercentage));
+          enum AmountKind {
+            WITH_FEE,
+            WITHOUT_FEE,
           }
 
-          const fee = amount.mul(swapFeePercentage).div(fp(1));
-          const protocolFee = fee.mul(protocolFeePercentage).div(fp(1));
-          return protocolFee;
-        }
-
-        context('on swaps given in', () => {
-          it('tracks fees when swapping tokens', async () => {
-            const previousDueFee = await pool.getDueProtocolFeeBptAmount();
+          function getApproxDueFee(amount: BigNumber, kind: AmountKind, recoveryMode: boolean): BigNumber {
```

Comment:
```suggestion
          function getExpectedProtocolFee(amount: BigNumber, kind: AmountKind, recoveryMode: boolean): BigNumber {
```

I'd rename this function (along with the `approxFee` variables to `expectedProtocolFee`) so that the tests are of the format `expect(currentX).to.be..equalWithError(expectedX, 0.01)`